enumerables.rb

```diff
@@ -0,0 +1,129 @@
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |item|
+      new_array << item if yield(item) == true
+    end
+    new_array.empty? ? self : new_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return false unless yield(item)
+      elsif pattern.nil?
+        return false unless item
+      else
+        return false unless pattern === item
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return true unless yield(item) == false
+      elsif pattern.nil?
+        return true unless item
+      else
+        return true unless (pattern === item) == false
+      end
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false].my_any? == [nil, false].any? => false
```
- Yours returns the wrong answer
![image](https://user-images.githubusercontent.com/52209067/81322093-ab10a680-9050-11ea-84f5-3f567039c832.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/81322127-b95ec280-9050-11ea-8693-1125d4ff3cf1.png)